### PR TITLE
RT - work on normalising application state, code cleanup, typings, code reuse

### PR DESCRIPTION
**Pull request by zeala**

### CODE REVIEW

File: src/client/src/redux/actions/spotTileActions.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import { createAction } from 'redux-actions'
```
Comment (bhavesh-desai-scratch) [inline src/client/src/redux/actions/spotTileActions.ts]: We need to break this file up. It has Epics as well as actions

File: src/client/src/redux/reducers/combineReducers.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { combineReducers } from 'redux'
```
Comment (bhavesh-desai-scratch) [inline src/client/src/redux/reducers/combineReducers.ts]: We need to name this file better. Maybe root reducer

File: src/client/src/redux/reducers/currencyPairsReducer.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { CurrencyPair } from '../../types/index'
+import { ACTION_TYPES } from '../actions/referenceDataActions'
+
+interface CurrencyPairReducerState {
+  [id: string]: CurrencyPair
+}
+
+export const currencyPairReducer = (state: CurrencyPairReducerState = {}, action): CurrencyPairReducerState => {
+  switch (action.type) {
```
Comment (bhavesh-desai-scratch) [inline src/client/src/redux/reducers/currencyPairsReducer.ts]: Better to use a map then a switch

File: src/client/package.json [hunk 4]
```diff
@@ -87,6 +87,7 @@
     "redux-observable": "0.17.0",
     "redux-persist": "5.6.5",
     "redux-persist-crosstab": "3.7.0",
+    "reselect": "3.0.1",
```
Comment (dwalker11) [inline src/client/package.json]: 👍

**Pull request merged**